File: Fraud_Detection/src/main/java/com/example/fraud/model/FraudAlert.java
package com.example.fraud.model;

public class FraudAlert {
    private String message;
    private Transaction transaction;

    public FraudAlert(String message, Transaction transaction) {
        this.message = message;
        this.transaction = transaction;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Transaction getTransaction() {
        return transaction;
    }

    public void setTransaction(Transaction transaction) {
        this.transaction = transaction;
    }
}

File: fraud-detection-frontend/src/components/FraudAlert.js
import React from 'react';

function FraudAlert({ alerts }) {
  return (
    <div>
      <h2>Fraud Alerts</h2>
      {alerts.length > 0 ? (
        <ul>
          {alerts.map((alert, index) => (
            <li key={index}>
              {alert.message} for transaction ID: {alert.transaction.id}
            </li>
          ))}
        </ul>
      ) : (
        <p>No fraud alerts</p>
      )}
    </div>
  );
}

export default FraudAlert;
